### PR TITLE
Forecasting using LSTM, hybrid models (AE, VAE) with and without classifier networks 

### PR DESCRIPTION
**Pull request by Saran-nns**
Preparing documentation. For the usability and experiments, please refer trajatest notebook.

### CODE REVIEW

**Review by justinshenk — APPROVED** [review]
LGTM

**Review by justinshenk — COMMENTED** [review]
LGTM

File: .travis.yml [hunk 1]
```diff
@@ -1,56 +1,35 @@
...
   provider: pypi
   user: jshenk
   skip_cleanup: true
-  skip_existing: true
   on:
     tags: true
     branch: master
-  password:
-    secure: o5ON/6Q4aORM4dgTVUQ39w0N+Gc+6Ala+K5J16b5lnNWGgHglqIlJzYXJo8THpeNYTm6ZbEDQEFurCTEKA/MZ2WzreePWQ4Z4E2dIihqhI+71rSbForRPKunV2CEr/QQdUEzXe6npO2UTnO0zDS5XMSrlBncKO4F4zUvrYTuXLj5fES0IFiFHMWxEpNaXMKiypfcRIKJriRbHY22/H8uSgzFluxRG+UqpbJz+R94bqIg30wBJw4nI9JMI00Du67eCO91t+aQ26+5Am+DqA6+jawd89OVPxtlLSdWtgtxPmWAD/IBLP2d7sqfK+QnezmH8NuAMB6DJdTkbscHcvYT8itHg8csBDdvfH8xoA9x8f+Cc60gviKaBoayORFF7FXkjyAYTCSfEi2dfxTTDR0UisbEG99k0+25+DMHxdC8z7/NQz4qal2vKfhPe8kTsOPQLwh0EHmdVU+v9M9LgrLhN55/lI/a6w+zL1/BJ6ZO6arMhHLVmgRtHP+Ckq6OKwQJYNwZxsg8PfwZxl0jFfd3yVX9lS9s95An90z9mEPheC8zQNz2fzAZUZun6GI9u/FCrGpMbrzKzq4R0UtNc8mfipHJ/v027+C2x43wkXA0c6Zvf9b7i6Bgm6EonnTagWrkQ0RdwqiKDd3smfgK2QZzD4G9vuv6z0w5CFhHL9v1Oc0=
+  password:
```
Comment (WolfByttner) [inline .travis.yml:34]: Are we supposed to check in the password?

File: traja/models/train.py [hunk 2]
```diff
@@ -0,0 +1,696 @@
...
+
+        self.model_optimizers = optimizer.get_optimizers(lr=self.lr)
+        self.model_lrschedulers = optimizer.get_lrschedulers(
+            factor=self.lr_factor, patience=self.scheduler_patience
+        )
+
+    def __str__(self):
+        return f"Training model type {self.model_type}"
+
+    def fit(self, train_loader, test_loader, model_save_path=None):
```
Comment (WolfByttner) [inline traja/models/train.py:148]: We should make epochs an argument to fit(). You might want to change it on the fly.

Also, I think we should inject the training mode. You sometimes want to alternate and experiment with different training strategies for the heads.

I created a pull request to that effect

File: traja/main.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+          TrajaDataFrame -- copy
+
+        """
+        data = self._data
+        if deep:
+            data = data.copy()
+        return TrajaDataFrame(data).__finalize__(self)
+
+
+# TODO: Replace with tests.
```
Comment (WolfByttner) [inline traja/main.py:133]: Dead code

File: .travis.yml (hunk 1 above)
Comment (justinshenk) [reply .travis.yml:34]: It’s encrypted so safe to include. The benefit is that pypi releases are updated on successful build. https://stackoverflow.com/a/57501840, https://medium.com/@mikkokotila/deploying-python-packages-to-pypi-with-travis-works-9a6597781556

**Comment by WolfByttner** [comment]
Commits merged in rebase https://github.com/traja-team/traja/pull/33
